cartographer_ros/cartographer_ros/msg_conversion.cc

```diff
@@ -404,10 +404,17 @@ std::unique_ptr<nav_msgs::OccupancyGrid> CreateOccupancyGridMsg(
       const uint32_t packed = pixel_data[y * width + x];
       const unsigned char color = packed >> 16;
       const unsigned char observed = packed >> 8;
-      const int value =
-          observed == 0
-              ? -1
-              : ::cartographer::common::RoundToInt((1. - color / 255.) * 100.);
+
+      int value = -1;
+      if (observed != 0) {
+        const auto cell = ::cartographer::common::RoundToInt((1. - color / 255.) * 100.);
+        if (publish_ros_map) {
+          if (cell < 49) value = 0;
```

Comment:
What' the idea behind this?

Reply:
The idea is to set the values into the trinary interpretation, so MoveBase can plan through this map, and the exploratory packages can find frontiers.

Reply:
Can you add this explanation into the code? What I don't understand is why those values and not others?